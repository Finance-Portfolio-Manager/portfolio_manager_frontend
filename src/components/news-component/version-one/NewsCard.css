/* $bg: #F9F9F9 */
/* $card: #fff */
@import url('https://fonts.googleapis.com/css?family=Arbutus+Slab|Open+Sans&display=swap');

.newspaper_body{
  color: #495456;
  margin: 72px;
  font-family: Open Sans, sans-serif;
}

.newspaper__h2{
  font-family: Arbutus Slab, serif;
  font-weight: normal;
  color: #000;
  line-height: 1.25;
}

.newspaper_text{
  position: absolute;
  bottom: 0;
  font-size: 14px;
}

.newspaper__container{
  max-width: 900px;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(160px, 1fr));
  grid-gap: 48px;
  margin: 0 auto;
 }
.newspaper__card{
  cursor: pointer;
  position: relative;
  height: 0;
  padding-bottom: 120%
  /* --offset-multiplier: 4px;
  &:hover
    --offset-multiplier: 6px
  transition: transform 0.6s ease
  --translate: 0
  transform: translate(var(--translate), var(--translate))
  &:hover
    --translate: calc(-1px * (var(--cards) - 1))
    transition: transform 0.3s ease} */
}
/* .child
  position: absolute
  width: 100%
  height: 100%
  padding: 0 16px
  box-sizing: border-box
  background: $card
  box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1), 0px -4px 8px rgba(255, 255, 255, 0.8)
  border-radius: 6px
  transition: inherit
  --translate: calc(var(--offset) * var(--offset-multiplier))
  transform: translate(var(--translate), var(--translate))
  z-index: 5

  @for $i from 1 through 5
    &:nth-child(#{$i})
      --offset: #{-1+$i}
      z-index: #{5 - $i}
   */