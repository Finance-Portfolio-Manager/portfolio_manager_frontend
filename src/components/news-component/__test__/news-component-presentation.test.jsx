import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import NewsCard from './../news-component-presentation';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const wrapper = mount(<NewsCard></NewsCard>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<NewsCard></NewsCard>).toJSON();
    expect(tree).toMatchSnapshot();
})

// test("1 card body renders", ()=>{
//     expect(wrapper.find("CardBody").length).toBe(1);
// })

// test("2 carousel controls render", ()=>{
//     expect(wrapper.find("CarouselControl").length).toBe(2);
// })

// describe("Performing actions on the news component", ()=>{
//     afterEach(cleanup);

//     test("click next", ()=>{
//         const nextControl = wrapper.find("CarouselControl").last();
//         const clickAction = ()=>{
//             nextControl.simulate('click');
//         }
//         expect(clickAction).not.toThrow();
//     })
    
//     test("click previous", ()=>{
//         const prevControl = wrapper.find("CarouselControl").first();
//         // console.log(prevControl.debug());
//         const clickAction = ()=>{
//             prevControl.simulate('click');
//         }
//         expect(clickAction).not.toThrow();
//     })

//     test("click carousel indicators", ()=>{
//         const carIndicator = wrapper.find("CarouselIndicators");
//         const clickAction = ()=>{
//             carIndicator.simulate('click');
//         }
//         expect(clickAction).not.toThrow();
//     })
// })