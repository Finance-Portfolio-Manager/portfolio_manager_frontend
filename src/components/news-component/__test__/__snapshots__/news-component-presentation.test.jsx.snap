// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="w-20 border border-0  card"
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="carousel slide"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="carousel-inner"
    />
    <a
      className="carousel-control-prev"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex="0"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="sr-only"
      />
    </a>
    <a
      className="carousel-control-next"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex="0"
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="sr-only"
      />
    </a>
  </div>
</div>
`;
